streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessorSupplier.java

```diff
@@ -198,7 +201,7 @@ private void emit(final TimeOrderedKeyValueBuffer.Eviction<K, V> toEmit) {
                 final ProcessorRecordContext prevRecordContext = internalProcessorContext.recordContext();
                 internalProcessorContext.setRecordContext(toEmit.recordContext());
                 try {
-                    internalProcessorContext.forward(toEmit.key(), toEmit.value());
+                    internalProcessorContext.forward(toEmit.record());
```

Comment:
I think this might actually be a bug. IIRC, the Record forward call overrides the context, so you might have to actually set all the fields in Record from the context when you construct it in `toEmit.record()`. Specifically, I think we might be dropping the headers here.

I might also be wrong, so you might want to double-check me first.

It looks like we were never testing whether we propagate headers through Suppress. If it turns out that we never were, then there's also no problem. The case to look out for is if we _were_, but now we no longer are.

Reply:
It does, yet. I just tested in one of the methods. Adding the change and updating a test to cover this.